pkgs/build-support/rust/import-cargo-lock.nix

```
@@ -89,18 +93,22 @@ let
     '';
     fetchurl {
       name = "crate-${pkg.name}-${pkg.version}.tar.gz";
-      url = "https://crates.io/api/v1/crates/${pkg.name}/${pkg.version}/download";
+      url = "${registryUrl}/v1/crates/${pkg.name}/${pkg.version}/download";
       sha256 = checksum;
     };
 
+  registries = {
+    "registry+https://github.com/rust-lang/crates.io-index" = "https://crates.io/api";
+  } // extraRegistries;
+
   # Fetch and unpack a crate.
   mkCrate = pkg:
     let
       gitParts = parseGit pkg.source;
     in
-      if pkg.source == "registry+https://github.com/rust-lang/crates.io-index" then
+      if builtins.hasAttr pkg.source registries then
       let
-        crateTarball = fetchCrate pkg;
+        crateTarball = fetchCrate (builtins.getAttr pkg.source registries) pkg;
```

this is more readable to me

```suggestion
        crateTarball = fetchCrate registries.${pkg.source} pkg;
```